chain/network/src/network_protocol/mod.rs

```diff
@@ -129,14 +130,15 @@ impl AccountKeySignedPayload {
     pub fn len(&self) -> usize {
         self.payload.len()
     }
+    pub fn signature(&self) -> &near_crypto::Signature {
+        &self.signature
+    }
     pub fn verify(&self, key: &PublicKey) -> bool {
         self.signature.verify(&self.payload, key)
     }
 }
 
-// TODO(gprusak): if we expect this to be large, perhaps we should
-// pass around an Arc, rather than pass it by value.
-#[derive(PartialEq, Eq, Clone, Debug, Hash)]
+#[derive(PartialEq, Eq, Debug, Hash)]
 pub struct SignedAccountData {
```

Comment:
This thing is effectively immutable (as it is signed), and we always pass an Arc of it. I think that this means that we probably should just put an `Arc` inside of it. Can be refactored separately from this PR.

Reply:
added a TODO